File: src/main/java/com/project/pantau/dto/report_status/ReportStatusResponse.java
package com.project.pantau.dto.report_status;

import com.project.pantau.enums.ReportStatus;
import com.project.pantau.enums.UserRole;
import lombok.Builder;

import java.time.LocalDateTime;
import java.util.UUID;

@Builder
public record ReportStatusResponse(
        UUID id,
        ReportStatus fromStatus,
        ReportStatus toStatus,
        String note,
        UserRole actorRole,
        LocalDateTime createdAt
) {
}

File: internal/dto/report_status/report_status_response.go
package reportstatus

import (
	"github.com/google/uuid"
	"pantau/internal/enums"
	"time"
)

type ReportStatusResponse struct {
	ID         uuid.UUID           `json:"id"`
	FromStatus *enums.ReportStatus `json:"from_status"`
	ToStatus   enums.ReportStatus  `json:"to_status"`
	Note       *string             `json:"note"`
	ActorRole  enums.UserRole      `json:"actor_role"`
	CreatedAt  time.Time           `json:"created_at"`
}
